app/src/main/java/net/harutiro/uwbanchorsystem/feature/serial/repository/SerialRepository.kt

```
@@ -0,0 +1,195 @@
+        var q = qIn
+
+        // 2の補数を計算（符号ビットの処理）
+        if ((qIn and (1 shl (bits - 1))) != 0) {
+            q -= (1 shl bits)
+        }
+
+        // 小数部に変換
+        val frac = q.toDouble() / (1 shl nFracs)
+
+        // 四捨五入して返す
+        return String.format("%.${roundOf}f", frac).toDouble()
```

detekt (1.23.8)

[warning] String.format("%.${roundOf}f", frac) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)